### ISSUE TITLE
[BUG] Labels in Popup don't show at all on Android

### ISSUE THREAD

**Issue by imsam67** [CommunityToolkit/Maui#1621]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

I've placed a `Label` as well as a `CollectionView` with `Label` in `ItemTemplate` inside a `Popup` control. On Android, I see none of the text I place in labels -- see screenshot. There should be a title displayed above the `CollectionView` inside the ``Popup`. There should also be fruit names inside the gray boxes/borders.

![Screenshot 2023-12-22 154225](https://github.com/CommunityToolkit/Maui/assets/15844030/34af2135-4d17-45dd-8298-9fef46bfd2cd)

I see another issue where they talk about a similar issue if the `Label` is within a `Frame`. In my case, I have `Label` controls both inside a `Border` as well as outside. None of the `Label`'s are showing anything.

### Expected Behavior

Labels should simply display the text.

### Steps To Reproduce

1. Add a new `Popup` to a .NET MAUI app
2. Place a `Label` control in the `Popup` making sure the color is set to something that will be visible

You can also download the code to the public repo I'm providing.

### Link to public reproduction project repository

https://github.com/iQuestLLC/test-maui

### Environment

```markdown
- .NET MAUI CommunityToolkit: 7.0.0
- OS: Android
- .NET MAUI: 8
```

**Comment by imsam67** [comment]
The problem seems to be caused by `VerticalTextAlignment`. Removing it makes the text visible, though alignment is still not right. Even though I set the `HorizontalTextAlignment` to `Center`, text is still not correctly centered but at least the text shows up.

**Comment by cat0363** [comment]
Regarding Android, there is a problem with the way the Measure method is called.
By applying PR #1520, you can get the following results:

<img width="300" src="https://github.com/CommunityToolkit/Maui/assets/125236133/06021405-9852-4408-9390-b24254a8b538" />

**Comment by vhugogarcia** [comment]
Thanks @imsam67 for reporting this issue. I'm closing this issue since this is a duplicate of https://github.com/CommunityToolkit/Maui/issues/1592, which covers the same topic.

**Comment by imsam67** [comment]
I just tested my app using `CommunityToolkit.Maui 99.0.0-preview1491`.  See the behavior in attached video.

Before using the preview version mentioned above, if I set the `VerticalTextAlign` to `Center`, text would not center. It also would not display as mentioned in original post.

Using the preview version however, setting `VerticalTextAlign` to `Center` seems to be working fine but there's another issue. As you'll see in attached video, initially `Label`'s don't seem to display anything but once I scroll up and down, text start to show.

In my code, I have a `CollectionView` inside a `Popup` and the `ItemTemplate` looks like this:

```
<CollectionView>
   ... 
   <CollectionView.ItemTemplate>
      <DataTemplate x:DataType="ShoppingListItem">
         <Border>
            <Grid
                ColumnDefinitions="50,*">
               <CheckBox
                   Grid.Column="0"
                   ... />
               <Label
                   Grid.Column="1"
                   ... />
            </Grid>
         </Border>
      </DataTemplate>
   </CollectionView.ItemTemplate>
</CollectionView>
```

Here's the behavior:


https://github.com/CommunityToolkit/Maui/assets/15844030/a555dae1-ad42-4f5e-8595-9d3c84be6916


https://github.com/CommunityToolkit/Maui/assets/15844030/869c93ef-dff1-40ac-aebe-cd64ecf055ce
